### PR TITLE
Ticks (handleTime) and collision handling

### PR DESCRIPTION
**Pull request by LukasK-roboyor**
- created method for updating position of the collision preview
- implemented handleTime() method
- added method that handles the game mechanics following a collision on the ground or a block while moving down
- added possibility for moving the tetromino down by 1
- completed rows now will be deleted

### CODE REVIEW

File: src/GameCollection/Tetris/Game.cpp [hunk 1]
```diff
@@ -3,7 +3,19 @@
```
Comment (Maze7) [inline src/GameCollection/Tetris/Game.cpp]: documentation is missing

File: src/GameCollection/Tetris/Game.cpp [hunk 2]
```diff
@@ -32,6 +44,61 @@ TetrisGame::Tetromino::TETROMINO_TYPE TetrisGame::Game::generateRandom()
 	return randomTetrominoType;
 }
```
Comment (Maze7) [inline src/GameCollection/Tetris/Game.cpp]: it is not really important, but it seems to be easier if methods be in same location and order as in the associated header

File: src/GameCollection/Tetris/Game.cpp [hunk 3]
```diff
@@ -32,6 +44,61 @@ TetrisGame::Tetromino::TETROMINO_TYPE TetrisGame::Game::generateRandom()
...
+	spawns a new tetromino.
+
+	Example usage:
+	if (!isPosValid())
+		handleCollision();
+*/
+void TetrisGame::Game::handleCollision()
+{
+	// Move it back into a valid position
+	m_currentTetromino.move(Tetromino::UP);
```
Comment (Maze7) [inline src/GameCollection/Tetris/Game.cpp]: this should be documentated and/or checked if it is needed to move up. It seems to be error-prone to just move the tetromino every method-call one position up without checking.  At least this behavior should be highlighted that this method should only called, if the current block is in an invalid position.

Something better would be. 

```
if (!isPosValid) {
  	// Move it back into a valid position
	m_currentTetromino.move(Tetromino::UP);
}
```

File: src/GameCollection/Tetris/Game.cpp [hunk 4]
```diff
@@ -32,6 +44,61 @@ TetrisGame::Tetromino::TETROMINO_TYPE TetrisGame::Game::generateRandom()
...
+	// Add the tetromino to the grid
+	m_playfield->addTetromino(m_currentTetromino);
+
+	// Check for completed rows and delete them
+	std::vector<int> completedRows = m_playfield->checkForCompletedRows();
+	for (int i = 0; i < completedRows.size(); i++)
```
Comment (Maze7) [inline src/GameCollection/Tetris/Game.cpp]: maybe do it with for-each like loop. Its would be  easier to understand on the first look

File: src/GameCollection/Tetris/Game.cpp (hunk 3 above)
Comment (LukasK-roboyor) [reply src/GameCollection/Tetris/Game.cpp]: changed in 68ffc3dd857b3024b6a2466a7d29fe229f46309b

File: src/GameCollection/Tetris/Game.cpp (hunk 4 above)
Comment (LukasK-roboyor) [reply src/GameCollection/Tetris/Game.cpp]: changed in 68ffc3dd857b3024b6a2466a7d29fe229f46309b

File: src/GameCollection/Tetris/Game.cpp (hunk 1 above)
Comment (Maze7) [reply src/GameCollection/Tetris/Game.cpp]: will added later

**Pull request merged**